src/lib/dbus.ts

```diff
@@ -1,37 +1,50 @@
-import * as Bluebird from 'bluebird';
 import * as dbus from 'dbus';
+import { promisify } from 'util';
 import { TypedError } from 'typed-error';
+import _ = require('lodash');
 
 import log from './supervisor-console';
 
 export class DbusError extends TypedError {}
 
-const bus = dbus.getBus('system');
-const getInterfaceAsync = Bluebird.promisify(bus.getInterface, {
-	context: bus,
-});
+const getBus = _.memoize((dbusLib: typeof dbus) => dbusLib.getBus('system'));
```

Comment:
The issue with that is that in a test environment, mocha loads all module imports from all spec files before starting tests. When dbus is imported, it gets a system bus immediately (import with side effect) which would cause the test to fail (see test failures in https://github.com/balena-os/balena-supervisor/pull/1961/commits/b775bf3aee7c6b2e6fff9ed6a27532c9b9d4c95b) due to not finding the bus socket in the test env. 

The old mocked-dbus implementation got around this by stubbing `dbus.getBus` on import as well (another import with side effect), but the downside to this was that the mocked dbus module was hard to parse.

If you'd prefer to keep the old implementation in `src/lib/dbus.ts` (i.e. not use memoizee), I can look into this more.

Reply:
An example of when src/lib/dbus is imported is in spec 41, the v1 API tests.

Reply:
I imagine you do it because you want to mock the interface, but maybe you can try if it is really necessary? Adding the memoization here and in `getInterfaceAsync` will add additional overhead to these calls. While is probably negligible, it still better to make sure we really need it.

Reply:
After looking more into it, the additional overhead added by removing the imports with side effects was not worth the added complexity.. instead, I documented why `dbus.getBus` is stubbed at the root level, and added some missing methods to the stub: https://github.com/balena-os/balena-supervisor/pull/1961/files#diff-eb84a8c34f22bf98ab3a2fff02e87b96d60104060f13bc2954e83ab85d0265d2R5-R17